tslib.es6.js

```diff
@@ -185,11 +185,23 @@ export function __makeTemplateObject(cooked, raw) {
     return cooked;
 };
 
+var __getImportStarCache = function () {
+    if (typeof WeakMap !== "function") return null;
+
+    var cache = new WeakMap();
+    __getImportStarCache = function () { return cache; }
+    return cache;
+};
+
 export function __importStar(mod) {
     if (mod && mod.__esModule) return mod;
+    if (mod === null || (typeof mod !== "object" && typeof mod !== "function")) return { default: mod }

```

Comment:
Why should a `function` be treated as a module object? Also, would we not also want to cache the `{ default: mod }` result as well?

Reply:
If you moved this line above the previous line, the previous line could be shortened to `if (mod.__esModule) return mod;`. There's also a semicolon missing at the end of the line.

Reply:
> Why should a `function` be treated as a module object?

So that you can do:
```js
import EventEmitter, { getEventListeners } from "events";

export function getListenersOrNull(target) {
	if (target instance EventEmitter) {
		return getEventListeners(target);
	}

	if (target instanceof EventTarget) {
		try {
			return getEventListeners(target);
		} catch {}
	}

	return null;
}
```

which gets transpiled to:
```js
"use strict";
exports.getListenersOrNull = getListenersOrNull;
var events = __importStar(require("events"));

function getListenersOrNull(target) {
	if (target instance events.default) {
		return events.getEventListeners(target);
	}

	if (target instanceof EventTarget) {
		try {
			return events.getEventListeners(target);
		} catch {}
	}

	return null;
}
```

Reply:
> Also, would we not also want to cache the `{ default: mod }` result as well?

That’s not possible because `WeakMap` only accepts `object`s.

---

Also, this fast path ensures that:
```js
import * as hasSymbolsNS from "has-symbols";

if (hasSymbolsNS.default === true) {
	// ...
}
```

keeps working when transpiled to:

```js
var hasSymbolsNS = __importStar(require("has-symbols"));

if (hasSymbolsNS.default === true) {
	// ...
}
```